core/src/test/scala/unit/kafka/server/AbstractFetcherThreadTest.scala

```diff
@@ -171,25 +171,25 @@ class AbstractFetcherThreadTest {
 
       // Now check message's crc
       val records = partitionData.toRecords
-      for (entry <- records.shallowEntries.asScala) {
-        entry.record.ensureValid()
-        logEndOffset = entry.nextOffset
+      for (batch <- records.batches.asScala) {
+        batch.ensureValid()
+        logEndOffset = batch.nextOffset
       }
     }
 
     override protected def fetch(fetchRequest: DummyFetchRequest): Seq[(TopicPartition, TestPartitionData)] = {
       fetchCount += 1
       // Set the first fetch to get a corrupted message
       if (fetchCount == 1) {
-        val corruptedRecord = Record.create("hello".getBytes())
-        val badChecksum = (corruptedRecord.checksum + 1 % Int.MaxValue).toInt
-        // Garble checksum
-        ByteUtils.writeUnsignedInt(corruptedRecord.buffer, Record.CRC_OFFSET, badChecksum)
-        val records = MemoryRecords.withRecords(corruptedRecord)
+        val record = new SimpleRecord("hello".getBytes())
+        val records = MemoryRecords.withRecords(CompressionType.NONE, record)
+        ByteUtils.writeUnsignedInt(records.buffer, 15, Int.MaxValue)
```

Comment:
I'm not sure we need a dependence on the offset here. Maybe we can just flips some arbitrary bits in the payload?